Documentation/benchmarks/etcd-3-watch-memory-benchmark.md

```diff
@@ -0,0 +1,72 @@
+# Watch Memory Usage Benchmark
+
+*NOTE*: The watch features are still being actively developed and we expect the memory usage to change over time. However we do expect that memory usage should not be significantly increased from the figures below.
+
+A primary goal of etcd is to support large numbers of watching. etcd aims to support as many as O(10k) clients, O(100K) watch streams (O(10) streams per client) and O(10M) total watchings (O(100) watching per stream). Thus the goal is to minimize the memory consumption per watching, since it will dominate the overall consumption.
```

Comment:
A primary goal of etcd is supporting a very large number of watchers doing a massively large amount of watching. etcd aims to support O(10k) clients, O(100K) watch streams (O(10) streams per client) and O(10M) total watchings (O(100) watching per stream). The memory consumed by each individual watching accounts for the largest portion of etcd's overall usage, and is therefore the focus of current and future optimizations.

(PS -- for my own edification, what does the notation "O(10k)" represent here? And it that's theta, do we want a theta character?)